### PR TITLE
Implement accessors for `OpamProcess.command` type

### PR DESCRIPTION
**Pull request by Leonidas-from-XIV**
This is useful when wanting to implement other runners for `OpamProcess.Job.Op`, e.g. based on Dune's Fibers, Lwt or Async as it allows to run the process using their process spawning abstractions.

### CODE REVIEW

**Review by kit-ty-kate — APPROVED** [review]

File: src/core/opamProcess.ml
Comment (rjbou) [inline src/core/opamProcess.ml:219]: Please use same naming than existing
```suggestion
let cmd_of_command c = c.cmd
let args_of_command c = c.args
let dir_of_command c = c.cmd_dir
let env_of_command c = c.cmd_env
let allow_stdin_of_command c = c.cmd_stdin
```

File: src/core/opamProcess.mli
Comment (rjbou) [inline src/core/opamProcess.mli:49]:
```suggestion
val command_cmd : command -> string
val command_args : command -> string list
val command_dir : command -> string option
val command_env : command -> string array option
val command_allow_stdin : command -> bool option
```

**Review by rjbou — CHANGES_REQUESTED** [review]
Ok on the idea, comme comments/fix

File: src/core/opamProcess.ml
Comment (kit-ty-kate) [reply src/core/opamProcess.ml]: I'm not sure about using this naming convention for those accessors. The `*_of_*` functions in OCaml usually imply some kind of processing (e.g. conversion from one type to the other)

Accessors are more straightforward. Furthermore I think it isn't clear for users what those do without reading the implementation, especially for `cmd_of_command` it isn't clear to me from the name what the difference between `cmd_of_command` and `string_of_command` is. `command_cmd` looks more like the typical conversion for accessors to me.

**Review by rjbou — APPROVED** [review]

File: src/core/opamProcess.mli
Comment (Leonidas-from-XIV) [reply src/core/opamProcess.mli]: Thanks! I think this was due me vendoring opam and dune apparently just believed the interfaces without building the implementations. Will be careful about this in the future!

File: src/core/opamProcess.ml
Comment (Leonidas-from-XIV) [reply src/core/opamProcess.ml]: I would generally agree, which is why originally I had them using the more usual accessor "convention". In the end I am ok with whatever OPAM maintainers prefer.

Comment (rjbou) [reply src/core/opamProcess.ml]: Good question. The idea behind my proposal is to keep it homogeneous with existing ones.  We'll end up with `text_of_command` and `command_args`, both direct accessors. 
`string_of_` is another naming convention, in opam it is always used to have the full string of a type (with Module.to_string)

Comment (kit-ty-kate) [reply src/core/opamProcess.ml]: The only existing one is `text_of_command`, the rest are not accessors. With that in mind i think it's fine to have two naming conventions.

Comment (Leonidas-from-XIV) [reply src/core/opamProcess.ml:219]: I've pushed a separate commit renaming them, it can be removed if desired before merging.

**Comment by Leonidas-from-XIV** [comment]
@kit-ty-kate I tried it and that works as well for me. I didn't expose it originally because exposing a type can be painful later if the type is to be changed.

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**
